coalahtml/tree.py

```diff
@@ -0,0 +1,84 @@
+from collections import OrderedDict
+from queue import Queue
+import os
+
+
+class Node:
+    """
+    Node class
+    """
+    count = 1
+
+    def __init__(self, path):
+        self.child = {}
+        self.parent = None
+        self.path = path
+        self.abspath = ""
+        self._id = Node.count
+        Node.count += 1
+
+
+class Tree:
+
+    def __init__(self, file_list):
+        """
+        Create a root and build the graph.
+        """
+        self.root = Node(os.path.commonprefix(file_list))
+        self.root.abspath = self.root.path
+        self.fl_split = Tree._split_paths(file_list)
+        self._build(self.fl_split)
+
+    @staticmethod
+    def _split_paths(path_list):
+        return [path.split('/') for path in path_list]
+
+    def _add_node(self, new_node, prev_node, head):
+        q = Queue()
+        q.put(head)
+
+        while not q.empty():    # Search for parent node: BFS
+            current = q.get()
+            if current._id != prev_node._id:
+                for child in current.child.values():
+                    q.put(child)
+            else:
+                prev_node = current
```

Comment:
To convince me, just give me the test case where before and after `prev_node` is different.